### ISSUE TITLE
RF-DETR-Seg models mAP drops significantly after conversion to OV / ONNX

### ISSUE THREAD

**Issue by leoll2** [open-edge-platform/training_extensions#7154]
**Steps to reproduce**

1. Create an instance segmentation project, e.g. importing [this dataset](http://storage.geti.intel.com/test-data/geti/datasets/qa/fire-coco-medium.zip)
2. Train a RF-DETR-Seg model of any size
3. Check the reported mAP for the Pytorch, ONNX and OpenVINO variants.

**Expected**

All variants achieve a similar score, since the conversion to OV / ONNX shouldn't significantly affect the model performance.

**Actual**

The OpenVINO and ONNX variants show a significantly lower score than PyTorch.

<img width="911" height="165" alt="Image" src="https://github.com/user-attachments/assets/1cd6bb1f-a2d5-43c7-a49e-6270ed83832f" />

<img width="907" height="161" alt="Image" src="https://github.com/user-attachments/assets/381d4503-03d7-46c5-ba60-e20d13d7455a" />

<img width="914" height="242" alt="Image" src="https://github.com/user-attachments/assets/0840fc0a-3db2-46fe-a7bf-c17b735aa908" />

Nevertheless, predictions (generated through OV) are not bad at all, so it could only an evaluation issue.

<img width="1010" height="736" alt="Image" src="https://github.com/user-attachments/assets/b2583752-7783-4f4a-b439-49c767cb8aed" />

_Note:_ the problem only affects RF-DETR models for **instance segmentation**, but not RF-DETR for **object detection**.

**Comment by leoll2** [comment]
The root cause was a bug in Model API (#7190), which caused predicted labels to be misaligned w.r.t. to the ground truth ones by 1, thus ruining the mAP. Fixed in Model API v0.4.6.
